### PR TITLE
[To Main] Task - DESENG-676: Combine custom and summary content

### PR DESCRIPTION
**Pull request by NatSquared**
Issue #: [🎟️ DESENG-676](https://citz-gdx.atlassian.net/browse/DESENG-676)
**Description of changes:**
- **Task** Merge engagement summary content and custom content 
  - Combine the text and rich content fields from redundant types SummaryContent and CustomContent and store them in
    the existing EngagementContent table
  - Update the API to handle the new content structure
  - Todo: Update the frontend to properly leverage the new content structure and allow editing of the simplified content type
**User Guide update ticket (if applicable):**

### CODE REVIEW

File: met-web/src/components/engagement/old-view/EngagementContent.tsx [hunk 1]
```diff
@@ -5,16 +5,16 @@ import { ActionContext } from './ActionContext';
 import { Skeleton } from '@mui/material';
 import { getEditorStateFromRaw } from 'components/common/RichTextEditor/utils';
 
-export const EngagementContent = () => {
-    const { richContent, isEngagementLoading } = useContext(ActionContext);
+export const EngagementContent = ({ index = 0 }: { index?: number }) => {
```
Comment (Baelx) [inline met-web/src/components/engagement/old-view/EngagementContent.tsx:8]: Thanks for making the old view work with this. Hopefully we can ditch this code soon

File: met-web/tests/unit/components/engagement/old-engagement.test.tsx [hunk 2]
```diff
@@ -187,7 +192,7 @@ describe('Engagement View page tests', () => {
         mockWidgetsRtkUnwrap.mockReturnValueOnce(Promise.resolve([whoIsListeningWidget]));
         mockContactRtkUnwrap.mockReturnValueOnce(Promise.resolve(mockContact));
         const { container } = render(<RouterProvider router={router} />);
-
+        screen.debug(undefined, 10000);
```
Comment (Baelx) [inline met-web/tests/unit/components/engagement/old-engagement.test.tsx:195]: Does this need to be here still?

**Review by Baelx — APPROVED** [review]
Great work on this! I'm always happy to see files being deleted in the code base :)

There may be a test debugging function that needs to be removed!

File: met-web/src/components/engagement/old-view/EngagementContent.tsx (hunk 1 above)
Comment (NatSquared) [reply met-web/src/components/engagement/old-view/EngagementContent.tsx:8]: I hope so too 😆

**Pull request merged**